From 136dd493681a624d980ddcbcf949e306e5328b27 Mon Sep 17 00:00:00 2001
From: Ryan Zhang <ryan.zhang@nxp.com>
Date: Tue, 16 Dec 2025 18:02:58 +0800
Subject: [PATCH] fix: Initialize all handle which will free in destructor
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the random crash caused by free a non‑null garbage value.
This patch is the improvement of PR:#1453

Upstream-Status: Submitted [https://github.com/KhronosGroup/Vulkan-Samples/pull/1453]
Signed-off-by: Ryan Zhang <ryan.zhang@nxp.com>
---
 .../api/timestamp_queries/timestamp_queries.h | 38 +++++++++----------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/samples/api/timestamp_queries/timestamp_queries.h b/samples/api/timestamp_queries/timestamp_queries.h
index 376ef17..b4b6861 100644
--- a/samples/api/timestamp_queries/timestamp_queries.h
+++ b/samples/api/timestamp_queries/timestamp_queries.h
@@ -64,17 +64,17 @@ class TimestampQueries : public ApiVulkanSample
 
 	struct
 	{
-		VkPipeline skybox;
-		VkPipeline reflect;
-		VkPipeline composition;
-		VkPipeline bloom[2];
+		VkPipeline skybox      = VK_NULL_HANDLE;
+		VkPipeline reflect     = VK_NULL_HANDLE;
+		VkPipeline composition = VK_NULL_HANDLE;
+		VkPipeline bloom[2]    = {VK_NULL_HANDLE};
 	} pipelines;
 
 	struct
 	{
-		VkPipelineLayout models;
-		VkPipelineLayout composition;
-		VkPipelineLayout bloom_filter;
+		VkPipelineLayout models       = VK_NULL_HANDLE;
+		VkPipelineLayout composition  = VK_NULL_HANDLE;
+		VkPipelineLayout bloom_filter = VK_NULL_HANDLE;
 	} pipeline_layouts;
 
 	struct
@@ -87,17 +87,17 @@ class TimestampQueries : public ApiVulkanSample
 
 	struct
 	{
-		VkDescriptorSetLayout models;
-		VkDescriptorSetLayout composition;
-		VkDescriptorSetLayout bloom_filter;
+		VkDescriptorSetLayout models       = VK_NULL_HANDLE;
+		VkDescriptorSetLayout composition  = VK_NULL_HANDLE;
+		VkDescriptorSetLayout bloom_filter = VK_NULL_HANDLE;
 	} descriptor_set_layouts;
 
 	// Framebuffer for offscreen rendering
 	struct FrameBufferAttachment
 	{
-		VkImage        image;
-		VkDeviceMemory mem;
-		VkImageView    view;
+		VkImage        image = VK_NULL_HANDLE;
+		VkDeviceMemory mem   = VK_NULL_HANDLE;
+		VkImageView    view  = VK_NULL_HANDLE;
 		VkFormat       format;
 		void           destroy(VkDevice device)
 		{
@@ -109,20 +109,20 @@ class TimestampQueries : public ApiVulkanSample
 	struct FrameBuffer
 	{
 		int32_t               width, height;
-		VkFramebuffer         framebuffer;
+		VkFramebuffer         framebuffer = VK_NULL_HANDLE;
 		FrameBufferAttachment color[2];
 		FrameBufferAttachment depth;
-		VkRenderPass          render_pass;
-		VkSampler             sampler;
+		VkRenderPass          render_pass = VK_NULL_HANDLE;
+		VkSampler             sampler     = VK_NULL_HANDLE;
 	} offscreen;
 
 	struct
 	{
 		int32_t               width, height;
-		VkFramebuffer         framebuffer;
+		VkFramebuffer         framebuffer = VK_NULL_HANDLE;
 		FrameBufferAttachment color[1];
-		VkRenderPass          render_pass;
-		VkSampler             sampler;
+		VkRenderPass          render_pass = VK_NULL_HANDLE;
+		VkSampler             sampler     = VK_NULL_HANDLE;
 	} filter_pass;
 
 	std::vector<std::string> object_names;
-- 
2.43.0

